src/elements/content-sidebar/MetadataSidebarRedesign.tsx

```diff
@@ -70,25 +71,43 @@ function MetadataSidebarRedesign({
     const { formatMessage } = useIntl();
 
     const [selectedTemplates, setSelectedTemplates] = React.useState<Array<MetadataTemplate>>([]);
-    const [editingTemplate, setEditingTemplate] = React.useState<MetadataTemplateInstance | MetadataTemplate | null>(
-        null,
-    );
+    const [editingTemplate, setEditingTemplate] = React.useState<MetadataTemplateInstance | null>(null);
     const [isUnsavedChangesModalOpen, setIsUnsavedChangesModalOpen] = React.useState<boolean>(false);
 
-    const { file, templates, errorMessage, status, templateInstances } = useSidebarMetadataFetcher(
-        api,
-        fileId,
-        onError,
-        isFeatureEnabled,
-    );
+    const { createMetadataInstance, file, templates, errorMessage, status, templateInstances, updateMetadataInstance } =
+        useSidebarMetadataFetcher(api, fileId, onError, isFeatureEnabled);
 
     const handleUnsavedChanges = () => {
         setIsUnsavedChangesModalOpen(true);
     };
 
+    const convertToInstance = (template: MetadataTemplate): MetadataTemplateInstance => {
+        return {
+            canEdit: !!file.permissions.can_upload,
```

Comment:
is it same for current version of MetadataSidebar?

Reply:
I brought that from `createMetadata` where we check the permission         
`const canEdit = !!permissions.can_upload` and than pass it to `createEditor` / `createTemplateInstance`, but if we want to convert template into instance on select, we need to set `canEdit`